projects/igniteui-angular/src/lib/grids/grid-public-cell.ts

```diff
@@ -60,6 +60,21 @@ export class IgxGridCell implements CellType {
         return this._column || this.grid.getColumnByName(this._columnField);
     }
 
+    private searchColumnIndexByName(columnIndex: number, name: string): any {
+        const columns = this.grid.getAllColumnsByName(name);
+
+        if (columns.length === 0 || columnIndex < 0 || columnIndex >= columns.length) {
+            return -1;
+        }
+    
+        const columnAtIndex = columns[columnIndex];
```

Comment:
columnIndex here is a "global" index, i.e. corresponding to column position among all grid columns. Using this index to query a column from a filtered columns collction (line 64), is not correct and will return an error when index exceeds collection length.